src/ape/managers/_contractscache.py

```diff
@@ -561,41 +561,26 @@ def get(
 
```

Comment:
@fubuloubu So I have been looking at this, and on lines 558 - 560, the logic is kinda weird and I am wondering your thoughts on changing it. Right now, if a contract type exists and you give it a default, it overrides the contract type and uses the default. IMO that is a strange way to update a contract type; there are better ways.

Here is my thought: What if instead, we combine the ABIs in this case too... You give it a default, there is a cached contract-type, the default is different than the cached, so combine them.

Also, I'd be down for removing that part, but I fear it would be a breaking change. By combing the ABIs it avoids being as break-y

Reply:
I think merging them might work, worst case they can just nuke the file from the cache folder.

A practical example is my `ape-tokens` plugin, which loads everything from the tokenlist as a generic `ERC20` contract type. I think right now it does not set `fetch_from_explorer=False` so it actually does take advantage of this behavior that loads from the explorer the more "full" type, but I could imagine where that kwarg was added to the section of the `ape-tokens` plugin so that it always loads _at least_ an ERC20 (with or without an explorer), *if* a cached type doesn't exist from the user previously calling `Contract` directly with `fetch_from_explorer=True`, where the type was merged according to this suggested new behavior.